### ISSUE TITLE
[FEATURE] Keyword Tracking

### ISSUE THREAD

**Issue by Cahllagerfeld** [EddieHubCommunity/api#188]
### ⭐ Suggestion

Like mentioned in [this](https://github.com/EddieHubCommunity/EddieBot/issues/598) Feature-Request, we should store special Keywords and the user who subscribed to the Keyword in the Database

### 💻 Use Cases

Community Partners could get a ping when a special  word is posted within a message.

**Comment by vedanthv** [comment]
This is a nice idea! What tech stack are your preferring to work on this?

**Comment by Cahllagerfeld** [comment]
The API uses Nestjs and Datastax Astra / Stargate as database

**Comment by BOLT04** [comment]
can you give an example of how the feature would be used @Cahllagerfeld ?

**Comment by Cahllagerfeld** [comment]
In my opinion the bot needs to send the message to the api. The api checks if the message contains a tracked word, and in case it is, it needs to send the userids back to the bot. The bot then sends a dm with the message URL to the UserIds it got from the api. Does this make sense? For now its more like an idea, I didnt spend any efforts in implementing this feature yet, so there might be some issues with the realization.

**Comment by BOLT04** [comment]
yes that helps @Cahllagerfeld, thank you 👍. I agree with sending the whole message, because the bot doesn't know what keywords there are.
With that said, we should be using `HTTPS` for sure, since we would be sending user's data through the network 👍. Not sure any additional layer on top of HTTPS is needed to secure the user's message.... but I think nothing else is needed in terms of protocols. For the API one thing that is crucial is authorization and authentication, only the bot can use this endpoint to get userIds and the message URL. Otherwise, other apps can ping users...

**Comment by BOLT04** [comment]
With this feature, we also need the ability to unsubscribe to keyword tracking, much like the open source reminders and other reminder features 😃. The user that gets pinged should have a link or instructions on how to unsubscribe, or subscribe to more keywords

**Comment by Cahllagerfeld** [comment]
>
>only the bot can use this endpoint to get userIds and the message URL. Otherwise, other apps can ping users...
>

I already implemented a mechanism to identify 'trusted' and 'regular' apps doing the request to the api

**Comment by BOLT04** [comment]
awesome!! Do you have that implementation in some branch or PR @Cahllagerfeld ? Or is it documented somewhere in the repo? 
Sorry, I'm not familiar with the API at its current state 😅

**Comment by Cahllagerfeld** [comment]
Its already merged in the main-branch. You can take a look at the [ascii-docs](https://eddiehubcommunity.github.io/api/) I implemented.

**Comment by Cahllagerfeld** [comment]
The API is able to sign access-tokens with different scopes (Read and Write). Only trusted apps, which have a Secret are allowed to create Write-Tokens. We can check for the scopes if they are trusted or not.

**Comment by BOLT04** [comment]
thank you 👍 I had a look at the docs and we might need to add a new scope for this new endpoint, right?

**Comment by eddiejaoude** [comment]
The API authentication is in the process of being simplified
